pwndbg/commands/search.py

```diff
@@ -126,8 +126,19 @@ def search(type, hex, string, executable, writable, value, mapping, save, next):
 
     # Null-terminate strings
     elif type == 'string':
+        value = value.encode()
         value += b'\x00'
 
+    # Find the mappings that we're looking for
+    mappings = pwndbg.vmmap.get()
+
+    if mapping:
+        mappings = [m for m in mappings if mapping in m.objfile]
+
+    if not mappings:
+        print(M.red("Could not find mapping %r" % mapping))
```

Comment:
Also if we want to show 'errors' in red, we should probably do that with all of them